src/logic/nfts/marketplace.ts

```diff
@@ -0,0 +1,255 @@
+      return 'name'
+    case NFTSortBy.RECENTLY_LISTED:
+      return 'searchOrderCreatedAt'
+    case NFTSortBy.CHEAPEST:
+      return 'searchOrderPrice'
+    default:
+      return getMarketplaceOrderBy(NFT_DEFAULT_SORT_BY)
+  }
+}
+
+export function fromMarketplaceNFTFragment(
+  fragment: MarketplaceNFTFragment
+): NFTResult {
+  const result: NFTResult = {
+    nft: {
+      id: getId(fragment.contractAddress, fragment.tokenId),
+      tokenId: fragment.tokenId,
+      contractAddress: fragment.contractAddress,
+      activeOrderId:
+        fragment.activeOrder && !isExpired(fragment.activeOrder.expiresAt)
+          ? fragment.activeOrder.id
+          : null,
+      owner: fragment.owner.address.toLowerCase(),
+      name: fragment.name || capitalize(fragment.category),
+      image: fragment.image || '',
+      url: `/contracts/${fragment.contractAddress}/tokens/${fragment.tokenId}`,
+      data: {
+        parcel: fragment.parcel
+          ? {
+              description:
+                (fragment.parcel.data && fragment.parcel.data.description) ||
+                null,
+              x: fragment.parcel.x,
+              y: fragment.parcel.y,
+              estate: fragment.parcel.estate
+                ? {
+                    tokenId: fragment.parcel.estate.tokenId,
+                    name:
+                      (fragment.parcel.estate.data &&
+                        fragment.parcel.estate.data.name) ||
+                      capitalize(NFTCategory.ESTATE),
+                  }
+                : null,
+            }
+          : undefined,
+        estate: fragment.estate
+          ? {
+              description:
+                (fragment.estate.data && fragment.estate.data.description) ||
+                null,
+              size: fragment.estate.size,
+              parcels: fragment.estate.parcels.map(({ x, y }) => ({
+                x: +x,
+                y: +y,
+              })),
+            }
+          : undefined,
+        wearable: fragment.wearable
+          ? {
+              bodyShapes: fragment.wearable.bodyShapes,
+              category: fragment.wearable.category,
+              description: fragment.wearable.description,
+              rarity: fragment.wearable.rarity,
+            }
+          : undefined,
+        ens: fragment.ens ? { subdomain: fragment.ens.subdomain } : undefined,
+      },
+      issuedId: null,
+      itemId: null,
+      category: fragment.category,
+      network: Network.ETHEREUM,
+      chainId: getMarketplaceChainId(),
+      createdAt: +fragment.createdAt * 1000,
+      updatedAt: +fragment.updatedAt * 1000,
+    },
+    order:
+      fragment.activeOrder && !isExpired(fragment.activeOrder.expiresAt)
+        ? fromMarketplaceOrderFragment(fragment.activeOrder)
+        : null,
+  }
+
+  // remove undefined data
+  for (const property of Object.keys(result.nft.data)) {
+    const key = property as keyof typeof result.nft.data
+    if (!result.nft.data[key]) {
```

Comment:
Should this be checked against undefined directly? Just asking in case we want to keep `0`s and `false`s